tests/search/includes/classes/test-class-versioning.php

```diff
@@ -514,6 +690,82 @@ public function activate_version_data() {
 					),
 				),
 			),
+
+			// "next" alias
+			array(
+				// Input array of versions
+				array(
+					2 => array(
+						'number' => 2,
+						'active' => true,
+					),
+					3 => array(
+						'number' => 3,
+						'active' => false,
+					),
+					4 => array(
+						'number' => 4,
+						'active' => false,
+					),
+				),
+				// Indexable slug
+				'post',
+				// Version to activate
+				'next',
```

Comment:
This test and the one below are failing because the version number is not normalized. Maybe it's not worth doing it in `activate_version()` (leaving normalization to upper layers), but it has to be done for sure in `test_activate_version()`, because apart from checking the expected new versions array, it also checks that the actually activated version (which is always a number) matches the version we want (and here it's not normalized):

```
$version_to_activate = self::$version_instance->normalize_version_number( $indexable, $version_to_activate );
```

Reply:
Ok I've fixed up the tests - once we actually integrate this into the other functions, we'll have to come back to these.